Libs/GObject/Classes/Object.ClosureHelper.cs

```diff
@@ -16,12 +16,13 @@ private class ClosureHelper : IDisposable
             // prevent the delegate from being collected by the GC
             private readonly ClosureMarshal? _marshalCallback;
             private readonly ActionRefValues? _callback;
+            private readonly Closure.ClosureSafeHandle _safeHandle; 
             
             #endregion
             
             #region Properties
-            
-            public IntPtr Handle { get; private set; }
+
+            public IntPtr Handle => _safeHandle?.DangerousGetHandle() ?? IntPtr.Zero;
```

Comment:
It might be worth checking if the handle is invalid before returning.

From the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle.dangerousgethandle?view=net-5.0#System_Runtime_InteropServices_SafeHandle_DangerousGetHandle):

> Using the DangerousGetHandle method can pose security risks because, **if the handle has been marked as invalid with SetHandleAsInvalid, DangerousGetHandle still returns the original**, potentially stale handle value. The returned handle can also be recycled at any point. At best, this means the handle might suddenly stop working. At worst, if the handle or the resource that the handle represents is exposed to untrusted code, this can lead to a recycling security attack on the reused or returned handle.

I think it should be sufficient to return null if the handle is invalid.

Reply:
Yes I'm going to change it as it is making the intend clearer.

I think it does improve security only marginally. If I understood the docs correctly the root problem is that we should not expose the intptr at all but only the safe handle.

So this PR improves only the memory management aspect . Every user of a GOject could safe it's handle (as it's a struct) and access it after the object was disposed / freed.

I'm going to open a new issue to verify/ improve the security aspect as part of the new generator.